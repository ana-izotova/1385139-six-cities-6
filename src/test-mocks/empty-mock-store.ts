import {AuthorizationStatus, Cities, FetchStatus, SortType, NameSpace} from "../const";
import {UserInitialStateTypes} from "../store/user/user-types";
import {AllOffersInitialStateTypes} from "../store/all-offers/all-offers-types";
import {SingleOfferInitialStateTypes} from "../store/single-offer/single-offer-types";
import {FavoritesInitialStateTypes} from "../store/favorites/favorites-types";

const userEmptyState: UserInitialStateTypes = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  login: ``,
  userAvatar: ``,
  fetchStatus: FetchStatus.INIT
};

const allOffersEmptyState: AllOffersInitialStateTypes = {
  currentCity: Cities[0],
  allOffers: [],
  currentSort: SortType.POPULAR,
  isDataLoaded: false,
  fetchStatus: FetchStatus.INIT,
  favoritesHaveBeenChanged: false
};

const singleOfferEmptyState: SingleOfferInitialStateTypes = {
  offer: null,
  offersNearby: [],
  comments: [],
  isOfferLoaded: false,
  fetchStatus: FetchStatus.INIT
};

const favoritesEmptyState: FavoritesInitialStateTypes = {
  favoriteCards: [],
  areFavoriteCardsLoaded: false
};

export const emptyMockStore = {
  [NameSpace.ALL_OFFERS]: allOffersEmptyState,
  [NameSpace.SINGLE_OFFER]: singleOfferEmptyState,
  [NameSpace.USER]: userEmptyState,
  [NameSpace.FAVORITES]: favoritesEmptyState
};
