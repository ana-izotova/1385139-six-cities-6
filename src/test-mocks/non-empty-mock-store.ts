import {AuthorizationStatus, Cities, FetchStatus, NameSpace, SortType} from "../const";
import {UserInitialStateTypes} from "../store/user/user-types";
import {SingleOfferInitialStateTypes} from "../store/single-offer/single-offer-types";
import {currentOffer, offersNearby, comments, allOffers, favoriteOffers} from "./adapted-data-mock";
import {AllOffersInitialStateTypes} from "../store/all-offers/all-offers-types";
import {FavoritesInitialStateTypes} from "../store/favorites/favorites-types";

const userStoreAuthorized: UserInitialStateTypes = {
  authorizationStatus: AuthorizationStatus.AUTH,
  login: `jondoe@mail.com`,
  userAvatar: `avatar.jpg`,
  fetchStatus: FetchStatus.DONE
};

const userStoreUnauthorized: UserInitialStateTypes = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  login: ``,
  userAvatar: ``,
  fetchStatus: FetchStatus.DONE
};

const singleOfferStore: SingleOfferInitialStateTypes = {
  offer: currentOffer,
  offersNearby,
  comments,
  isOfferLoaded: true,
  fetchStatus: FetchStatus.DONE,
  error: null
};

const allOffersStore: AllOffersInitialStateTypes = {
  currentCity: Cities[0],
  allOffers,
  currentSort: SortType.POPULAR,
  isDataLoaded: true,
  fetchStatus: FetchStatus.DONE,
  favoritesHaveBeenChanged: true,
  error: null
};

const favoritesStore: FavoritesInitialStateTypes = {
  favoriteCards: favoriteOffers,
  areFavoriteCardsLoaded: true
};

export const nonEmptyMockStoreUnauthorized = {
  [NameSpace.ALL_OFFERS]: allOffersStore,
  [NameSpace.SINGLE_OFFER]: singleOfferStore,
  [NameSpace.USER]: userStoreUnauthorized,
  [NameSpace.FAVORITES]: favoritesStore
};

export const nonEmptyMockStoreAuthorized = {
  [NameSpace.ALL_OFFERS]: allOffersStore,
  [NameSpace.SINGLE_OFFER]: singleOfferStore,
  [NameSpace.USER]: userStoreAuthorized,
  [NameSpace.FAVORITES]: favoritesStore
};

