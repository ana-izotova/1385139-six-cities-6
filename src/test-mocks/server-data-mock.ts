export const testOffer = {
  "city": {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13,
    },
  },
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  "images": [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": false,
  "is_premium": true,
  "rating": 4,
  "type": `room`,
  "bedrooms": 2,
  "max_adults": 2,
  "price": 500,
  "goods": [`Breakfast`, `Air conditioning`],
  "host": {
    "id": 25,
    "name": `Madelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-madelina.jpg`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "location": {
    "latitude": 48.83861,
    "longitude": 2.350499,
    "zoom": 16,
  },
  "id": 1,
};

export const testOffer2 = {
  "city": {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13,
    },
  },
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  "images": [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": false,
  "is_premium": true,
  "rating": 4,
  "type": `room`,
  "bedrooms": 2,
  "max_adults": 2,
  "price": 500,
  "goods": [`Breakfast`, `Air conditioning`],
  "host": {
    "id": 25,
    "name": `Madelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-madelina.jpg`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "location": {
    "latitude": 48.83861,
    "longitude": 2.350499,
    "zoom": 16,
  },
  "id": 2,
};

export const testComment = {
  "id": 1,
  "user": {
    "id": 15,
    "is_pro": false,
    "name": `Kendall`,
    "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
  },
  "rating": 3,
  "comment": `We loved it so much, the house, the veiw, the location just great.. Highly recommend :)`,
  "date": `2021-02-12T08:04:28.647Z`,
};

export const testOfferFavorited = {
  "city": {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13,
    },
  },
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  "images": [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": true,
  "is_premium": true,
  "rating": 4,
  "type": `room`,
  "bedrooms": 2,
  "max_adults": 2,
  "price": 500,
  "goods": [`Breakfast`, `Air conditioning`],
  "host": {
    "id": 25,
    "name": `Madelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-madelina.jpg`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "location": {
    "latitude": 48.83861,
    "longitude": 2.350499,
    "zoom": 16,
  },
  "id": 1,
};

export const testOffer2Favorited = {
  "city": {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13,
    },
  },
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  "images": [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": true,
  "is_premium": true,
  "rating": 4,
  "type": `room`,
  "bedrooms": 2,
  "max_adults": 2,
  "price": 500,
  "goods": [`Breakfast`, `Air conditioning`],
  "host": {
    "id": 25,
    "name": `Madelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-madelina.jpg`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "location": {
    "latitude": 48.83861,
    "longitude": 2.350499,
    "zoom": 16,
  },
  "id": 2,
};


