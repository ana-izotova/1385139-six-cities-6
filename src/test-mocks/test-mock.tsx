import React, {Dispatch} from 'react';
import configureStore from 'redux-mock-store';
import * as redux from 'react-redux';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import thunk, {ThunkDispatch} from 'redux-thunk';
import {RootStateType} from "../store/root-reducer";
import {ActionTypes} from "../store/action-types";
import {redirect} from "../store/middlewares/redirect";
import {emptyMockStoreAuthorized, emptyMockStoreUnauthorized} from "./empty-mock-store";
import {nonEmptyMockStoreUnauthorized, nonEmptyMockStoreAuthorized} from "./non-empty-mock-store";
import {AppThunk} from "../store/api-actions";

interface TestProps {
  children?: any,
  pushUrl?: string,
  emptyStore: boolean,
  authorized: boolean
}

export const TestMock: React.FC<TestProps> = ({children, pushUrl, emptyStore, authorized}) => {
  jest.spyOn(redux, `useDispatch`);
  jest.spyOn(redux, `useSelector`);

  const mockStore = configureStore<RootStateType, ThunkDispatch<RootStateType, AppThunk, ActionTypes>>([thunk, redirect]);

  let mock;
  if (emptyStore) {
    mock = authorized ? emptyMockStoreAuthorized : emptyMockStoreUnauthorized;
  } else {
    mock = authorized ? nonEmptyMockStoreAuthorized : nonEmptyMockStoreUnauthorized;
  }

  const store = mockStore(mock);
  const history = createMemoryHistory();

  if (pushUrl) {
    history.push(pushUrl);
  }

  return (
    <redux.Provider store={store}>
      <Router history={history}>
        {children}
      </Router>
    </redux.Provider>
  );
};


