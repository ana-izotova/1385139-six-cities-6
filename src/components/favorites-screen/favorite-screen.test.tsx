import React from 'react';
import {render} from '@testing-library/react';
import {TestMock} from "../../test-mocks/test-mock";
import FavoritesScreen from "./favorites-screen";

describe(`Favorite screen should render correctly`, () => {
  it(`Should render empty favorites list when no favorites`, () => {
    const {container} = render(
        <TestMock emptyStore={true} authorized={true} >
          <FavoritesScreen />
        </TestMock>
    );

    expect(container).toMatchSnapshot();
  });

  it(`Should render favorites list correctly`, () => {
    const {container} = render(
        <TestMock emptyStore={false} authorized={true} >
          <FavoritesScreen />
        </TestMock>
    );

    expect(container).toMatchSnapshot();
  });

  it(`Should show loader screensaver component if favorite cards are not loaded`, () => {
    const {container} = render(
        <TestMock emptyStore={true} authorized={false} >
          <FavoritesScreen />
        </TestMock>
    );

    expect(container).toMatchSnapshot();
  });
});
