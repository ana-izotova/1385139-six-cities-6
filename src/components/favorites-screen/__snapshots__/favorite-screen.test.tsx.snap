// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite screen should render correctly Should render empty favorites list when no favorites 1`] = `
<div>
  <div
    class="page"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link "
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                src="img/logo.svg"
                style="width: 81px; height: 41px;"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                    style="background-image: url(avatar.jpg); border-radius: 50%;"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    jondoe@mail.com
                  </span>
                </a>
              </li>
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/"
                  style="margin-left: 10px;"
                >
                  <span
                    class="header__user-name user__name"
                  >
                    Exit
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--favorites"
    >
      <div
        class="page__favorites-container container"
      >
        <section
          class="favorites favorites--empty"
        >
          <h1
            class="visually-hidden"
          >
            Favorites (empty)
          </h1>
          <div
            class="favorites__status-wrapper"
          >
            <b
              class="favorites__status"
            >
              Nothing yet saved.
            </b>
            <p
              class="favorites__status-description"
            >
              Save properties to narrow down search or plan your future trips.
            </p>
          </div>
        </section>
      </div>
    </main>
    <footer
      class="footer container"
    >
      <a
        class="footer__logo-link"
        href="/"
      >
        <img
          alt="6 cities logo"
          class="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>
</div>
`;

exports[`Favorite screen should render correctly Should render favorites list correctly 1`] = `
<div>
  <div
    class="page"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link "
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                src="img/logo.svg"
                style="width: 81px; height: 41px;"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                    style="background-image: url(avatar.jpg); border-radius: 50%;"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    jondoe@mail.com
                  </span>
                </a>
              </li>
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/"
                  style="margin-left: 10px;"
                >
                  <span
                    class="header__user-name user__name"
                  >
                    Exit
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--favorites"
    >
      <div
        class="page__favorites-container container"
      >
        <section
          class="favorites"
        >
          <h1
            class="favorites__title"
          >
            Saved listing
          </h1>
          <ul
            class="favorites__list"
          >
            <li
              class="favorites__locations-items"
            >
              <div
                class="favorites__locations locations locations--current"
              >
                <div
                  class="locations__item"
                >
                  <a
                    class="locations__item-link"
                    href="#"
                  >
                    <span>
                      Paris
                    </span>
                  </a>
                </div>
              </div>
              <div
                class="favorites__places"
              >
                <article
                  class="favorites__card place-card"
                >
                  <div
                    class="place-card__mark"
                  >
                    <span>
                      Premium
                    </span>
                  </div>
                  <div
                    class="favorites__image-wrapper place-card__image-wrapper"
                  >
                    <a
                      href="/offer/1"
                    >
                      <img
                        alt="Place image"
                        class="place-card__image"
                        height="110"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
                        width="150"
                      />
                    </a>
                  </div>
                  <div
                    class="favorites__card-info place-card__info"
                  >
                    <div
                      class="place-card__price-wrapper"
                    >
                      <div
                        class="place-card__price"
                      >
                        <b
                          class="place-card__price-value"
                        >
                          €
                          169
                        </b>
                        <span
                          class="place-card__price-text"
                        >
                          / night
                        </span>
                      </div>
                      <button
                        class="place-card__bookmark-button place-card__bookmark-button--active  button"
                        type="button"
                      >
                        <svg
                          class="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            xlink:href="#icon-bookmark"
                          />
                        </svg>
                        <span
                          class="visually-hidden"
                        >
                          To bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      class="place-card__rating rating"
                    >
                      <div
                        class="place-card__stars rating__stars"
                      >
                        <span
                          style="width: 60%;"
                        />
                        <span
                          class="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      class="place-card__name"
                    >
                      <a
                        href="/offer/1"
                      >
                        The house among olive 
                      </a>
                    </h2>
                    <p
                      class="place-card__type"
                    >
                      Room
                    </p>
                  </div>
                </article>
              </div>
            </li>
            <li
              class="favorites__locations-items"
            >
              <div
                class="favorites__locations locations locations--current"
              >
                <div
                  class="locations__item"
                >
                  <a
                    class="locations__item-link"
                    href="#"
                  >
                    <span>
                      Brussels
                    </span>
                  </a>
                </div>
              </div>
              <div
                class="favorites__places"
              >
                <article
                  class="favorites__card place-card"
                >
                  <div
                    class="favorites__image-wrapper place-card__image-wrapper"
                  >
                    <a
                      href="/offer/2"
                    >
                      <img
                        alt="Place image"
                        class="place-card__image"
                        height="110"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg"
                        width="150"
                      />
                    </a>
                  </div>
                  <div
                    class="favorites__card-info place-card__info"
                  >
                    <div
                      class="place-card__price-wrapper"
                    >
                      <div
                        class="place-card__price"
                      >
                        <b
                          class="place-card__price-value"
                        >
                          €
                          310
                        </b>
                        <span
                          class="place-card__price-text"
                        >
                          / night
                        </span>
                      </div>
                      <button
                        class="place-card__bookmark-button place-card__bookmark-button--active  button"
                        type="button"
                      >
                        <svg
                          class="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            xlink:href="#icon-bookmark"
                          />
                        </svg>
                        <span
                          class="visually-hidden"
                        >
                          To bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      class="place-card__rating rating"
                    >
                      <div
                        class="place-card__stars rating__stars"
                      >
                        <span
                          style="width: 96%;"
                        />
                        <span
                          class="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      class="place-card__name"
                    >
                      <a
                        href="/offer/2"
                      >
                        Waterfront with extraordinary view
                      </a>
                    </h2>
                    <p
                      class="place-card__type"
                    >
                      Apartment
                    </p>
                  </div>
                </article>
              </div>
            </li>
            <li
              class="favorites__locations-items"
            >
              <div
                class="favorites__locations locations locations--current"
              >
                <div
                  class="locations__item"
                >
                  <a
                    class="locations__item-link"
                    href="#"
                  >
                    <span>
                      Cologne
                    </span>
                  </a>
                </div>
              </div>
              <div
                class="favorites__places"
              >
                <article
                  class="favorites__card place-card"
                >
                  <div
                    class="place-card__mark"
                  >
                    <span>
                      Premium
                    </span>
                  </div>
                  <div
                    class="favorites__image-wrapper place-card__image-wrapper"
                  >
                    <a
                      href="/offer/3"
                    >
                      <img
                        alt="Place image"
                        class="place-card__image"
                        height="110"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"
                        width="150"
                      />
                    </a>
                  </div>
                  <div
                    class="favorites__card-info place-card__info"
                  >
                    <div
                      class="place-card__price-wrapper"
                    >
                      <div
                        class="place-card__price"
                      >
                        <b
                          class="place-card__price-value"
                        >
                          €
                          182
                        </b>
                        <span
                          class="place-card__price-text"
                        >
                          / night
                        </span>
                      </div>
                      <button
                        class="place-card__bookmark-button place-card__bookmark-button--active  button"
                        type="button"
                      >
                        <svg
                          class="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            xlink:href="#icon-bookmark"
                          />
                        </svg>
                        <span
                          class="visually-hidden"
                        >
                          To bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      class="place-card__rating rating"
                    >
                      <div
                        class="place-card__stars rating__stars"
                      >
                        <span
                          style="width: 84%;"
                        />
                        <span
                          class="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      class="place-card__name"
                    >
                      <a
                        href="/offer/3"
                      >
                        Penthouse, 4-5 rooms + 5 balconies
                      </a>
                    </h2>
                    <p
                      class="place-card__type"
                    >
                      Hotel
                    </p>
                  </div>
                </article>
              </div>
            </li>
          </ul>
        </section>
      </div>
    </main>
    <footer
      class="footer container"
    >
      <a
        class="footer__logo-link"
        href="/"
      >
        <img
          alt="6 cities logo"
          class="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>
</div>
`;

exports[`Favorite screen should render correctly Should show loader screensaver component if favorite cards are not loaded 1`] = `
<div>
  <div
    class="page page--gray page--main"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link "
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                src="img/logo.svg"
                style="width: 81px; height: 41px;"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/login"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      style="margin-left: auto; margin-right: auto; width: 20%;"
    >
      <h1>
        Loading, please wait...
      </h1>
      <div
        class="loader-container"
      >
        <div
          aria-busy="true"
          class=""
        >
          <svg
            aria-label="audio-loading"
            height="80"
            viewBox="0 0 38 38"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <lineargradient
                id="a"
                x1="8.042%"
                x2="65.682%"
                y1="0%"
                y2="23.865%"
              >
                <stop
                  offset="0%"
                  stop-color="#4481c3"
                  stop-opacity="0"
                />
                <stop
                  offset="63.146%"
                  stop-color="#4481c3"
                  stop-opacity=".631"
                />
                <stop
                  offset="100%"
                  stop-color="#4481c3"
                />
              </lineargradient>
            </defs>
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                transform="translate(1 1)"
              >
                <path
                  d="M36 18c0-9.94-8.06-18-18-18"
                  id="Oval-2"
                  stroke="#4481c3"
                  stroke-width="2"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="0.9s"
                    from="0 18 18"
                    repeatCount="indefinite"
                    to="360 18 18"
                    type="rotate"
                  />
                </path>
                <circle
                  cx="36"
                  cy="18"
                  fill="#fff"
                  r="1"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="0.9s"
                    from="0 18 18"
                    repeatCount="indefinite"
                    to="360 18 18"
                    type="rotate"
                  />
                </circle>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </main>
  </div>
</div>
`;
